### ISSUE TITLE
From absolute to relative IRI?

### ISSUE THREAD

**Issue by kjetilk** [kasei/perl-iri#15]
There is code to go from a relative IRI to an absolute IRI, but can we go the other way?

Say that we have e.g.
```perl
my $iri = IRI->new(value => "http://www.hestebedg\x{e5}rd.dk/hest/ku");
 ```
and then, from some other place, I get the base
```perl
my $base = IRI->new(value => "http://www.hestebedg\x{e5}rd.dk/hest");
 ```

Can I somehow get the relative URI (or just the string) `/ku`?

If not, then, that's on my wishlist :-)

**Comment by kasei** [comment]
That's on my wishlist, too. I started work on this in the rel branch, but I recall it having some complicated corner-cases that I wasn't able to address in the time I spent on it.

**Comment by kjetilk** [comment]
Oh, OK, I wasn't aware. I can't see any branches though, could you please push it here?

**Comment by kasei** [comment]
Oops!

**Comment by kasei** [comment]
Just a warning, I'm not sure where I left the code in the rel branch. There's some code, and tests, but no guarantees that the tests cover all cases, or that the code is the right approach! :)

**Comment by kjetilk** [comment]
OK, interesting! 

So, was the corner case of such a nature that it could break things, or just a suboptimal case?

**Comment by kasei** [comment]
I don't remember. I can look into the code tomorrow and try to figure out where I left things, but I can't do it tonight.

**Comment by kasei** [comment]
My main concern was avoiding a situation where I released code that did the *wrong* thing for the corner cases, without being able to identify them and at least fall back on bailing out and just using absolute IRIs.

**Comment by kjetilk** [comment]
Oh, great! Don't worry, it wasn't an immediate need, it just struck me as something that would have been a great enabler in many designs.

**Comment by kasei** [comment]
Agreed. I'd love to get this sorted out. I just have so many things fighting for my time at the moment!
